### PR TITLE
Fix false positives in Rails logger cop + add tests

### PR DESCRIPTION
**Pull request by shalvah-gs**
~Fixes the false positives in the cop that checks for missing `include Nxt::Loggable`. Controllers and maintenance tasks already include this module, so this should not raise an error.~

Update: this PR removes the cop entirely; as pointed out by @Aerdayne below, keeping it up to date may be an uphill battle.

This PR also adds some tests for the cops.

### CODE REVIEW

**Review by chiibis — APPROVED** [review]

File: custom_cops/nxt_core/rails/missing_logger_include.rb [hunk 1]
```diff
@@ -16,21 +16,44 @@ class MissingLoggerInclude < RuboCop::Cop::Base
         MSG = 'Missing `include Nxt::Loggable` for a class which uses the logger.'
 
         # NodePattern docs: https://github.com/rubocop/rubocop-ast/blob/master/docs/modules/ROOT/pages/node_pattern.adoc
+        # Test at https://nodepattern.herokuapp.com/
         def_node_matcher :includes_nxt_loggable?, <<~PATTERN
           (begin <(send nil? :include (const (const nil? :Nxt) :Loggable)) ...>)
         PATTERN
 
+        def_node_matcher :includes_nxt_core_loggable?, <<~PATTERN
+          (begin <(send nil? :include (const (const (const nil? :NxtCore) :Logging) :Loggable)) ...>)
```
Comment (Aerdayne) [inline custom_cops/nxt_core/rails/missing_logger_include.rb:25]: It's `Nxt::Core`, not `NxtCore`.

**Review by chiibis — APPROVED** [review]

**Review by Aerdayne — COMMENTED** [review]
I'm afraid that writing a cop for a thing like this is going to be an uphill battle. For example, ActiveJob classes also include the logger automatically, yet there's no whitelist entry for them in this PR. I think that having a cop that prevents people from using `Rails.logger` is enough. In case somebody forgets to add the logging module, shouldn't the tests fail anyway?

**Comment by shalvah-gs** [comment]
> In case somebody forgets to add the logging module, shouldn't the tests fail anyway?

Assuming the tests cover the logging. Sometimes it's in the sad path, where it may not be as thoroughly checked.

Still, all things considered, you may be right...Fixing a missing include error in production may involve less effort than maintaining this. 🤔

**Comment by shalvah-gs** [comment]
Removed the cop for missing includes 👍

**Review by Aerdayne — APPROVED** [review]

**Pull request merged**